CMakeLists.txt

```diff
@@ -0,0 +1,261 @@
+    src/processor/microdump.cc
+    src/processor/microdump_processor.cc
+    src/processor/minidump.cc
+    src/processor/minidump_processor.cc
+    src/processor/module_comparer.cc 
+    src/processor/module_serializer.cc
+    src/processor/pathname_stripper.cc
+    src/processor/proc_maps_linux.cc
+    src/processor/process_state.cc
+    src/processor/simple_symbol_supplier.cc
+    src/processor/source_line_resolver_base.cc
+    src/processor/stack_frame_cpu.cc
+    src/processor/stack_frame_symbolizer.cc
+    src/processor/stackwalker.cc
+    src/processor/stackwalker_address_list.cc
+    src/processor/stackwalker_amd64.cc
+    src/processor/stackwalker_arm.cc
+    src/processor/stackwalker_arm64.cc
+    src/processor/stackwalker_mips.cc
+    src/processor/stackwalker_ppc.cc
+    src/processor/stackwalker_ppc64.cc
+    src/processor/stackwalker_sparc.cc
+    src/processor/stackwalker_x86.cc
+    src/processor/symbolic_constants_win.cc
+    src/processor/tokenize.cc
+)
+
+set(LIBDISASM_SOURCES
+    src/third_party/libdisasm/ia32_implicit.c
+    src/third_party/libdisasm/ia32_implicit.h
+    src/third_party/libdisasm/ia32_insn.c
+    src/third_party/libdisasm/ia32_insn.h
+    src/third_party/libdisasm/ia32_invariant.c
+    src/third_party/libdisasm/ia32_invariant.h
+    src/third_party/libdisasm/ia32_modrm.c
+    src/third_party/libdisasm/ia32_modrm.h
+    src/third_party/libdisasm/ia32_opcode_tables.c
+    src/third_party/libdisasm/ia32_opcode_tables.h
+    src/third_party/libdisasm/ia32_operand.c
+    src/third_party/libdisasm/ia32_operand.h
+    src/third_party/libdisasm/ia32_reg.c
+    src/third_party/libdisasm/ia32_reg.h
+    src/third_party/libdisasm/ia32_settings.c
+    src/third_party/libdisasm/ia32_settings.h
+    src/third_party/libdisasm/libdis.h
+    src/third_party/libdisasm/qword.h
+    src/third_party/libdisasm/x86_disasm.c
+    src/third_party/libdisasm/x86_format.c
+    src/third_party/libdisasm/x86_imm.c
+    src/third_party/libdisasm/x86_imm.h
+    src/third_party/libdisasm/x86_insn.c
+    src/third_party/libdisasm/x86_misc.c
+    src/third_party/libdisasm/x86_operand_list.c
+    src/third_party/libdisasm/x86_operand_list.h
+)
+
+add_library(libbreakpad STATIC ${LIBBREAKPAD_SOURCES})
+add_library(libdisasm STATIC ${LIBDISASM_SOURCES} ${LIBDISASM_SOURCES})
+target_link_libraries(libbreakpad PRIVATE libdisasm)
+target_include_directories(libbreakpad
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+        $<INSTALL_INTERFACE:${include_install_dir}/breakpad>
+)
+add_definitions(
+    -DNOMINMAX
+    -DUNICODE
+    -DWIN32_LEAN_AND_MEAN
+    -D_CRT_SECURE_NO_WARNINGS
+    -D_CRT_SECURE_NO_DEPRECATE
+    -D_CRT_NONSTDC_NO_DEPRECATE
+    -D_LIBCPP_VERSION
+)
+
+if(WIN32)
+    if(WIN32)
+        file(GLOB_RECURSE LIBBREAKPAD_CLIENT_SOURCES src/client/windows/*.cc src/common/windows/*.cc)
+        include_directories("C:/Program Files (x86)/Microsoft Visual Studio 14.0/DIA SDK/include")
```

Comment:
Hardcoded path? Can we avoid it?

Reply:
> It is important to have this path

I didn't said that this path is not important, I'm saying about using some generalization. For example I have `DIA SDK` directory on my machine, but it's not in `C:/Program Files (x86)/Microsoft Visual Studio 14.0/DIA SDK` because I have nonstandard Visual Studio install folder.

Also having two paths from different Visual Studio versions unconditionally looks like a mistake for me.

Reply:
And if you add a variable, how to declare it later

Reply:
> And if you add a variable, how to declare it later

What variable to add? What declare later? I think I don't understand what you mean.